engine/order_manager.go

```diff
@@ -72,8 +73,12 @@ func (m *OrderManager) Start() error {
 	}
 	log.Debugln(log.OrderMgr, "Order manager starting...")
 	m.shutdown = make(chan struct{})
+	go func() {
+		m.processOrders()
+		m.run()
```

Comment:
Both being part of the same go routine technically still has the same issue, though it does alleviate startup. `m.processOrders()` has to finish before `m.run()` can begin and only via `m.run()`'s shutdown case does it execute `m.orderStore.wg.Done()` which allows GCT to shutdown with its `wg.Wait()`. Given the exact right circumstances, the application cannot shutdown until all orders for all exchanges have been processed

I've not encountered a situation where this is blocking. I think my fears were to do with the futures tracking which is not supported by any exchanges right now and that feature is much heavier in order cataloging. When that feature comes back, I may relook at this setup, but it's fine for now.

Reply:
My solution to this for the last few years has been a ctx passed with a bubble down cancel and everything can check ctx.Err() or add `<-ctx.Done()` to an existing select in order to cancel.
I'd have that anywhere like inside process Orders.
Everything is aware then if it needs to abort cleanly.

( And then add another context.WithCancel for actions lower down that need to be discretely cancelled )

Reply:
@gloriousCode So are you suggesting we goro both independently?

Reply:
I was just outlining the issues and potential solutions with my review approval, you don't need to do anything. Your code investigations demonstrate that I'll need to look more thoroughly at the order manager